src/test/java/test/integration/extension/AddDependencies.java

```diff
@@ -0,0 +1,20 @@
+package test.integration.extension;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines set of dependencies which will be added to deployed test archive.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AddDependencies {
+
+    /**
+     * List of maven artifact coordinates in the format of groupId:artifactId:version.
+     * @return
+     */
+    String[] value();
```

Comment:
For `@Environment` the reasoning, at least from my side, was not about errors - this you can always struggle with. It's more that I could see a reason to externalize keys to constants. Don't really see that for parts of `g:a:v`.